codenvy-api-factory/src/main/java/com/codenvy/api/factory/FactoryService.java

```diff
@@ -350,6 +350,77 @@ public void removeFactory(@ApiParam(value = "Factory ID", required = true)
         factoryStore.removeFactory(id);
     }
 
+
+    /**
+     * Updates factory with a new factory content
+     *
+     * @param id
+     *         id of factory
+     * @param newFactory
+     *         the new data for the factory
+     * @throws NotFoundException
+     *         when factory with given id doesn't exist
+     */
+    @ApiOperation(value = "Updates factory information by its ID",
+            notes = "Updates factory based on the Factory ID which is passed in a path parameter")
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "OK"),
+            @ApiResponse(code = 403, message = "User not authorized to call this operation"),
+            @ApiResponse(code = 404, message = "Factory not found"),
+            @ApiResponse(code = 500, message = "Internal Server Error")})
+    @POST
```

Comment:
my first intent was to use PUT (I think it's your "UPDATE" ? ) but @skabashnyuk told me there were problems using PUT so I fallback to POST

but if I can still use PUT it will be more in sync with REST API